### PR TITLE
some refactors, fixes and separations

### PR DESCRIPTION
**Pull request by stackerzzq**

### CODE REVIEW

File: command.c [hunk 1]
```diff
@@ -5,33 +5,33 @@
...
-static const int BUF_DELTER   = 2;
+static const int defaultBufSize = 32;
+static const int restBufSize    = 16;
+static const int bufDelter      = 1;
 
-void *nsq_buf_malloc(size_t buf_size, size_t n, size_t l)
+void *nsq_buf_malloc(size_t *buf_size, size_t n, size_t l)
 {
-    if (buf_size - n >= MIN_BUF_SIZE || buf_size - l >= MIN_BUF_SIZE) {
+    if ((int)(*buf_size - n) >= restBufSize || (int)(*buf_size - l) >= restBufSize) {
```
Comment (ploxiln) [inline command.c:14]: Can you put `nsq_buf_malloc()` and `nsq_buffer_add()` fixes into a separate pull request please?

Also, I'm still not sure this is right. Imagine `*buf_size == 64`, `n == 40`, and `l == 40`. In this case, this returns NULL  to mean no resize needed, but it needs to double the buffer size for `l` more bytes to fit after the existing `n` used bytes.

Comment (stackerzzq) [reply command.c:14]: Yes, it's definately not ok. All right, let me fix next time.

Comment (ploxiln) [reply command.c]: ... I would like to review and merge the `nsq_buffer_add()` changes first, and then other parts of this PR also in separate pull requests after that.

(I'm not a fan of git submodules ... in this case it's very small dependency, so  I had thought of just copying the sources into this repo, but I also have another idea to have the Makefile fetch the files only if they are out-of-date, I'll make a pull-request with that.)

Comment (stackerzzq) [reply command.c]: I just thought that there was no need to install libevbuffsock as a standard lib, as the connection socket was not so perfectly packed or no need to be packed into the libevbuffsock in my shallow opinion.

Comment (ploxiln) [reply command.c]: agreed, I just don't like  git submodules :)

**Pull request closed without merging**